src/sql-parser/src/ast/defs/expr.rs

```diff
@@ -816,6 +819,33 @@ pub struct Function<T: AstInfo> {
 
 impl<T: AstInfo> AstDisplay for Function<T> {
     fn fmt<W: fmt::Write>(&self, f: &mut AstFormatter<W>) {
+        // This block handles printing function calls that have special parsing. In stable mode, the
+        // name is quoted and so won't get the special parsing. We only need to print the special
+        // formats in non-stable mode.
+        if !f.stable() {
```

Comment:
This whole block is kinda janky because it only triggers in non-stable mode. I tried some other ideas but this one felt the least bad. If anyone has others would be happy to hear.

Reply:
From a code perspective everything looks good. I'm not sure what "stable mode" is for the parser though. When are we in stable mode and why do we only want to print the special formats in non-stable mode?

Reply:
When would we not be in stable mode? What I'm trying to understand are what are the real world scenarios in which we print these functions in a special way? Since it seems like we want both `extract(<arg>, <arg>)` and `extract(<arg> FROM <arg>)` to both be valid?

Reply:
`extract(<arg>, <arg>)` isn't valid because the `extract` keyword parses special, `"extract"(<arg>, <arg>)` however is valid because `"<x>"` parses an identifier with function arguments following, but doesn't care what the identifier is.

Normal mode is used for generating error messages and when generating statements that the postgres source executes.

Reply:
Ahhhh, I got it now, thank you for the context!